examples/pytorch/distributed_stl10/stl10_example.py

```diff
@@ -0,0 +1,360 @@
+# Inspired from the official ImageNet example: https://github.com/pytorch/examples/tree/master/imagenet
+
+import os
+import time
+import torch
+import socket
+import argparse
+import subprocess
+
+import torch.nn as nn
+import torch.distributed as dist
+import torchvision.transforms as transforms
+import torchvision.datasets as datasets
+import torchvision.models as models
+
+from typing import Tuple
+from torch.optim import SGD
+from torch.optim.optimizer import Optimizer
+from torch.utils.data import DataLoader
+from torch.utils.data.distributed import DistributedSampler
+from torch.nn.parallel import DistributedDataParallel
+
+
+class AverageMeter(object):
+    """Computes and stores the average and current value"""
+
+    def __init__(self):
+        self.reset()
+
+    def reset(self):
+        self.val = 0
+        self.avg = 0
+        self.sum = 0
+        self.count = 0
+
+    def update(self, val: float, n: int = 1):
+        self.val = val
+        self.sum += val * n
+        self.count += n
+        self.avg = self.sum / self.count
+
+
+def accuracy(output: torch.Tensor,
+             target: torch.Tensor,
+             topk: Tuple[int] = (1,)):
+    """Computes the precision@k for the specified values of k"""
+    with torch.no_grad():
+        maxk = max(topk)
+        batch_size = target.size(0)
+
+        _, pred = output.topk(maxk, 1, True, True)
+        pred = pred.t()
+        correct = pred.eq(target.view(1, -1).expand_as(pred))
+
+        res = []
+        for k in topk:
+            correct_k = correct[:k].view(-1).float().sum(0, keepdim=True)
+            res.append(correct_k.mul_(100.0 / batch_size))
+        return res
+
+
+def reduce_tensor(tensor: torch.Tensor, world_size: int):
```

Comment:
Also possible to use this https://github.com/pytorch/vision/blob/1aef87d01eec2c0989458387fa04baebcc86ea7b/references/detection/utils.py#L118 when there are many metrics to reduce

Reply:
Thanks! I wasn't aware of this.